pytest_stf/plugin.py

```diff
@@ -87,27 +92,35 @@ def fixture_allocated_phone(pytestconfig):
         ...
     }
     """
-    assert hasattr(pytestconfig, '_openstf'), 'stf_host or stf_token missing'
-
-    stf = pytestconfig._openstf  # pylint: disable=protected-access
-    stf  # type=StfClient
     requirements = pytestconfig.getoption('phone_requirements')
-    timeout = pytestconfig.getoption('stf_allocation_timeout')
     assert requirements, 'phone_requirements required'
-    requirements = parse_requirements(requirements)
+    assert 'platform' in requirements, 'platform required'
+
+    if hasattr(pytestconfig, '_openstf'):
+        stf = pytestconfig._openstf  # pylint: disable=protected-access
+        stf  # type=StfClient
+        timeout = pytestconfig.getoption('stf_allocation_timeout')
+        requirements = parse_requirements(requirements)
 
-    with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
-        yield device
+        with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
```

Comment:
timeout was actual allocation timeout when stf releases device internally unless client does it first. need to clarify timeout arg.